repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -324,6 +332,12 @@ mount -o "remount,rw" "$NEWROOT"
 
 ##### do the upgrade #######
 (
+    # check if leapp previously failed in the initramfs, if it did return to the emergency shell
+    [ -f "$NEWROOT$LEAPP_FAILED_FLAG_FILE" ] && {
+        echo "Leapp previously failed and cannot continue, returning back to emergency shell"
```

Comment:
One issue with the message is there are so many logging messages after it is printed that it is scrolled off screen.

Reply:
That is expected. Users should be ready to re-read the logs when debugging.